src/ui/components/common/ObservedBalance.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 @paritytech/contracts-ui authors & contributors
+// SPDX-License-Identifier: GPL-3.0-only
+
+import { useEffect, useState } from 'react';
+import { useApi } from '../../contexts';
+import { Balance } from 'types';
+
+const formatBalance = (balance: Balance, decimals = 12, unit?: string) => {
```

Comment:
Would be really nice to have some unit tests for this function, it's important that the result is accurate!

Reply:
Good point. I've added `vitest` as a unit test framework as `contracts-ui` didn't had any unit tests setup until now. Will probably come in handy more in the future.

https://github.com/paritytech/contracts-ui/pull/530/files#diff-30705ffdb2246bf3a84e24f11a573fc8eff793c3be5281de870072afe494856a

Reply:
Brilliant thank you :pray: